drop-in/src/main/java/com/adyen/checkout/dropin/ui/viewmodel/PaymentMethodsListDelegate.kt

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by atef on 21/10/2022.
+ */
+
+package com.adyen.checkout.dropin.ui.viewmodel
+
+import android.app.Application
+import com.adyen.checkout.components.ComponentAvailableCallback
+import com.adyen.checkout.components.base.Configuration
+import com.adyen.checkout.components.model.connection.OrderPaymentMethod
+import com.adyen.checkout.components.model.paymentmethods.PaymentMethod
+import com.adyen.checkout.components.model.paymentmethods.StoredPaymentMethod
+import com.adyen.checkout.components.model.payments.Amount
+import com.adyen.checkout.components.util.CurrencyUtils
+import com.adyen.checkout.components.util.PaymentMethodTypes
+import com.adyen.checkout.core.log.LogUtil
+import com.adyen.checkout.core.log.Logger
+import com.adyen.checkout.dropin.DropInConfiguration
+import com.adyen.checkout.dropin.R
+import com.adyen.checkout.dropin.checkPaymentMethodAvailability
+import com.adyen.checkout.dropin.ui.order.OrderModel
+import com.adyen.checkout.dropin.ui.paymentmethods.GiftCardPaymentMethodModel
+import com.adyen.checkout.dropin.ui.paymentmethods.PaymentMethodHeader
+import com.adyen.checkout.dropin.ui.paymentmethods.PaymentMethodListItem
+import com.adyen.checkout.dropin.ui.paymentmethods.PaymentMethodModel
+import com.adyen.checkout.dropin.ui.paymentmethods.PaymentMethodNote
+import com.adyen.checkout.dropin.ui.paymentmethods.StoredPaymentMethodModel
+import com.adyen.checkout.dropin.ui.stored.mapStoredModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+@Suppress("TooManyFunctions")
+internal class PaymentMethodsListDelegate constructor(
```

Comment:
The concept of delegates was only introduced for components and this is not a component. This delegate just adds an extra layer without a real purpose. I think it's better to have all this logic in `PaymentMethodsListViewModel`

Reply:
yeah good point we don't want to mistake viewModels with components